docs/spec_draft.md

```diff
@@ -522,6 +523,83 @@ it only exists to establish data dependencies from `result` to `inputs`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_reduce
+
+### Semantics
+
+Within each process group in the StableHLO grid, performs reduction, using
+`computations`, over the  values of the `operand` tensor from each process and
+produces a `result` tensor.
+
+The operation splits the StableHLO grid into process groups as follows:
+  * `channel_id <= 0` and `use_global_device_ids = false`,
```

Comment:
If `channel_id <= 0` and `use_global_device_ids = false`:
  ` process_groups` = `cross_replica(replica_groups)`

Reply:
would it be better to link the cross replica part of the doc?

Reply:
As a reader I would like to jump to cross replica from here and refer again how it works.

Reply:
Same feedback about linking to the parallel semantics section as in https://github.com/openxla/stablehlo/pull/576#discussion_r1035393503, i.e. I recommend that we don't. Otherwise, we'll need to link to a lot of stuff from a lot of ops, e.g. from Python snippets to individual ops.

"As a reader I would like to jump to cross replica from here and refer again how it works". A counterargument would be that this is a distracting detail for someone who already knows how those things work. Perhaps we should have a ticket to decide on the hyperlinking policy and discuss this separately from this PR.

Reply:
* If `channel_id <= 0` and `use_global_device_ids = false`:
      ` process_groups` = `cross_replica(replica_groups)`
      
would this be better way to write the same?